src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -73,6 +76,7 @@
           .put(CSHARP, CSharpTypeNameGenerator.class)
           .put(GO, GoTypeNameGenerator.class)
           .put(JAVA, JavaTypeNameGenerator.class)
+          .put(JS, NodeJSTypeNameGenerator.class) // Intentional mismatch.
```

Comment:
I wonder whether it makes sense to make switch the mismatch so that it's JSTypeNameGenerator.class, since NodeJS is a superset of JS.